core/actions/data_preparation.ts

```diff
@@ -128,6 +131,64 @@ function parseDataPreparationDefinitionJson(dataPreparationAsJson: {
   }
 }
 
+function resolveSourcesAndDestinations(
```

Comment:
Nit: rename to something like `applySessionToDataPreparationContents`

Reply:
Done.